kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryption.java

```diff
@@ -100,25 +105,85 @@ private static void checkCipherSuite() {
+        Long notFoundAliasExpireAfterWriteSeconds = getExperimentalLong(configuration.experimental(), "notFoundAliasExpireAfterWriteSeconds");
+        Long encryptionDekRefreshAfterWriteSeconds = getExperimentalLong(configuration.experimental(), "encryptionDekRefreshAfterWriteSeconds");
+        Long encryptionDekExpireAfterWriteSeconds = getExperimentalLong(configuration.experimental(), "encryptionDekExpireAfterWriteSeconds");
+        return new KmsCacheConfig(decryptedDekCacheSize, decryptedDekExpireAfterAccessSeconds, resolvedAliasCacheSize, resolvedAliasExpireAfterWriteSeconds,
+                resolvedAliasRefreshAfterWriteSeconds, notFoundAliasExpireAfterWriteSeconds, encryptionDekRefreshAfterWriteSeconds, encryptionDekExpireAfterWriteSeconds);
+    }
+
+    static DekManagerConfig dekManager(RecordEncryptionConfig configuration) {
+        Long maxEncryptionsPerDek = getExperimentalLong(configuration.experimental(), "maxEncryptionsPerDek");
+        return new DekManagerConfig(maxEncryptionsPerDek);
+
+    }
+
+    @Nullable
+    private static Integer getExperimentalInt(Map<String, Object> experimental, String property) {
+        return Optional.ofNullable(experimental).map(x -> x.get(property)).map(value -> {
+            if (value instanceof Number number) {
+                return number.intValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Integer.parseInt(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
+    @Nullable
+    private static Long getExperimentalLong(Map<String, Object> experimental, String property) {
+        return Optional.ofNullable(experimental).map(x -> x.get(property)).map(value -> {
+            if (value instanceof Number number) {
+                return number.longValue();
+            }
+            else if (value instanceof String stringValue) {
+                return Long.parseLong(stringValue);
+            }
+            else {
+                throw new IllegalArgumentException("could not convert " + property + " with type " + value.getClass().getSimpleName() + " to Integer");
+            }
+        }).orElse(null);
+    }
+
     @Override
     public SharedEncryptionContext<K, E> initialize(FilterFactoryContext context,
                                                     RecordEncryptionConfig configuration)
             throws PluginConfigurationException {
         checkCipherSuite();
-        KmsService<Object, K, E> kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
-        kmsPlugin.initialize(configuration.kmsConfig());
+        var kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
+        kmsPlugin.initialize(convertConfigValueToConfigType(kmsPlugin, configuration.kmsConfig()));
```

Comment:
could I get a for-dummies guide to what's going on here? 

We still have a `kms` field to id the plugin but are now working with a generic tree of Map/List/Values?

Reply:
One reason why this ^^ would be better than moving `convertConfigValueToConfigType` into the runtime it is means we're still using the same service loader/Jackson code for all plugins, which makes it easier to support both schema-first and java-first ways for a while (if we're lucky we can put off any major work till we've got the operator more developed)

Reply:
I went ahead and just implemented this, thus removing the hack. This came at the code of:

* A second Mojo, specifically for 'plugins'. 
* Adding `additionalProperties` in `SchemaObject`. We probably want to validate the schemas being consumed by the compiler.